pandas/core/frame.py

```diff
@@ -9805,7 +9805,8 @@ def join(
 
         Parameters
         ----------
-        other : DataFrame, Series, or list of DataFrame
+        other : DataFrame, Series, or a list containing any combination of DataFrames
+            and Series
```

Comment:
I see it now; the linter is complaining about the "and" being lowercase as if it's the start the description. I'd hazard a guess that sphinx does too. I think we should make the type more consise:

    DataFrame, Series, or a list of any combination of these